doc/basic_types.md

```diff
@@ -1,5 +1,360 @@
 # Basic Types
 
-You are reading a draft of the next documentation and it's in continuos update so
-if you don't find what you need please refer to:
-[C++ wrapper classes for the ABI-stable C APIs for Node.js](https://nodejs.github.io/node-addon-api/)
\ No newline at end of file
+Node Addon API consists of a few fundamental data types. These allow a user of 
+the API to create, convert and introspect fundamental JavaScript types, and
+interop with their C++ counterparts.
+
+## Value
+
+Value is the base class of Node Addon API's fundamental object type hierarchy.
+It represents a JavaScript value of an unknown type. It is a thin wrapper around
+the N-API datatype `napi_value`. Methods on this class can be used to check
+the JavaScript type of the underlying `napi_value` and also convert to C++ 
+types.
+
+### Constructor
+
+```cpp
+Value();                               
+```
+
+Used to create a Node Addon API `Value` that represents an **empty** value.
+
+```cpp
+Value(napi_env env, napi_value value); 
+```
+
+- `[in] env` - The `napi_env` environment in which to construct the Value object.
+- `[in] value` - The underlying JavaScript value that the `Value` instance represents
+
+Used to create a Node addon API `Value` that represents the `napi_value` passed
```

Comment:
Maybe here we need to use: Returns a Node addon API `Value` that represents the `napi_value` passed in.

Reply:
Agreed we to follow the standard format we should have a Returns after the parameters.